src/main/java/junmin/netflix/costreportservice/exception/InvalidProductionException.java

```diff
@@ -0,0 +1,14 @@
+package junmin.netflix.costreportservice.exception;
+
+public class InvalidProductionException extends Exception{
+	private int errorCode;
+
+	public InvalidProductionException(int errorCode, String message){
```

Comment:
Not quite understand "it easier to identify the error space for the codes". Are you saying using enum helps to identify the error type? 
In the application code, since the id is always passed in as errorEnum.getId(), so I feel like readability is sufficient. But let me know if you mean something else.

Reply:
Looking at just this class it is hard to know what values errorCode could take and what they would mean.  As you point out the developer could look at the application code to see how it is being used.  That's fine for a small application but it might be difficult for a larger application with lots of usages. Or if this code gets factored into a library then it might be difficult to find all the callers.

A typed enum instead of an integer would also prevent accidental usage.  E.g. a caller passing in an errorCode from a different enum.

Reply:
Points taken! Good consideration! Thank you!